pkg/service/controller.go

```diff
@@ -0,0 +1,68 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/instill-ai/model-backend/pkg/util"
+	controllerPB "github.com/instill-ai/protogen-go/vdp/controller/v1alpha"
+	modelPB "github.com/instill-ai/protogen-go/vdp/model/v1alpha"
+)
+
+func (s *service) GetResourceState(modelID string) (*modelPB.Model_State, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	resourceName := util.ConvertModelToResourceName(modelID)
+
+	resp, err := s.controllerClient.GetResource(ctx, &controllerPB.GetResourceRequest{
+		Name: resourceName,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Resource.GetModelState().Enum(), nil
+}
+
+func (s *service) UpdateResourceState(modelID string, state modelPB.Model_State, progress *int32, workflowId *string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	resourceName := util.ConvertModelToResourceName(modelID)
+
+	_, err := s.controllerClient.UpdateResource(ctx, &controllerPB.UpdateResourceRequest{
```

Comment:
@heiruwu nil: would be nice if we embed in the format: if _, err := xxx; err != nil {do something} to make src clean. There are some places we could improve also

Reply:
sure, I can update that